// nuxtify-api/categories/chart.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Chart = void 0;
const base_1 = require("../base");
class Chart extends base_1.Base {
    getHomeChart() {
        return __awaiter(this, void 0, void 0, function* () {
            try {
                const sig = this.createNoIdSig('/api/v2/page/get/chart-home');
                return yield this.createRequest('/api/v2/page/get/chart-home', {
                    sig,
                });
            }
            catch (err) {
                throw err;
            }
        });
    }
    getWeekChart(nationKey, week = 0, year = 0) {
        return __awaiter(this, void 0, void 0, function* () {
            const nation = {
                vn: 'IWZ9Z08I',
                kr: 'IWZ9Z0BO',
                us: 'IWZ9Z0BW',
            };
            const nationId = nation[nationKey];
            try {
                const sig = this.createIdSig('/api/v2/page/get/week-chart', nationId);
                return yield this.createRequest('/api/v2/page/get/week-chart', {
                    id: nationId,
                    year,
                    week,
                    sig,
                });
            }
            catch (err) {
                throw err;
            }
        });
    }
    getNewReleaseChart() {
        return __awaiter(this, void 0, void 0, function* () {
            try {
                const sig = this.createNoIdSig('/api/v2/page/get/newrelease-chart');
                return yield this.createRequest('/api/v2/page/get/chart-home', {
                    sig,
                });
            }
            catch (err) {
                throw err;
            }
        });
    }
}
exports.Chart = Chart;
